### PR TITLE
services/horizon: Remove db2/core.

### LINKED ISSUE #651: Horizon should not use core db anywhere outside ingest package
The description will be updated later. This issue now is only to track all the issues caused by connecting to core DB:
* https://github.com/stellar/go/issues/650
* https://github.com/stellar/go/issues/631 (slow, unoptimized queries made against core DB)
* https://github.com/stellar/go/issues/477
* https://github.com/stellar/go/issues/183

### PR DESCRIPTION
**Pull request by abuiles**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Remove db2/core.

### Why

Horizon shouldn't need access to core's database. 

Fix #651.

### CODE REVIEW

File: services/horizon/internal/init.go [hunk 1]
```diff
@@ -51,27 +50,6 @@ func mustInitHorizonDB(app *App) {
 	)}
 }
 
-func mustInitCoreDB(app *App) {
-	maxIdle := app.config.CoreDBMaxIdleConnections
-	maxOpen := app.config.CoreDBMaxOpenConnections
```
Comment (bartekn) [inline services/horizon/internal/init.go:56]: Remove these values from `Config` and `cmd/root.go`.

**Comment by 2opremio** [comment]
@abuiles can you merge this into `release-horizon-v1.6.0` once you are done?

File: services/horizon/internal/init.go (hunk 1 above)
Comment (abuiles) [reply services/horizon/internal/init.go:56]: @bartekn done. there were a couple of extra config values that we needed to removed.

**Comment by 2opremio** [comment]
I was hoping this would remove the requirement of setting a core URL when using captive-core, but I am still getting the following error:

```
Invalid config: stellar-core-db-url is blank. Please specify --stellar-core-db-url on the command line or set the STELLAR_CORE_DATABASE_URL environment variable.
```

**Comment by bartekn** [comment]
`--stellar-core-db-url ` is still required in `txsub` to submit transactions but we don't need it when not running a webserver (ex. in reingestion context). We should allow empty values when webserver is not running.

**Comment by 2opremio** [comment]
Addressed at https://github.com/stellar/go/pull/2780

**Comment by 2opremio** [comment]
Addressed at https://github.com/stellar/go/pull/2780

**Pull request merged**